src/realm/table.hpp

```diff
@@ -2131,7 +2143,8 @@ inline void Table::set_float(size_t col_ndx, size_t ndx, float value, bool is_de
 
 inline double Table::get_double(size_t col_ndx, size_t ndx) const noexcept
 {
-    return get<double>(col_ndx, ndx);
+    double d = get<double>(col_ndx, ndx);
+    return null::is_null_float(d) ? 0.0f : d;
```

Comment:
Should be `0.0`, even though I know it will be implicitly, trivially converted here. A case very similar to this has caused a bug in Realm (converting the `null` float representation to `double` and then back to `float`, which did not preserve the `null` value / bit pattern -  because of a ternary operator).